### ISSUE TITLE
Selection of specialised exception types

### ISSUE THREAD

**Issue by elfring** [breannansmith/scisim#1]
[Instances of the class "std::string" are thrown](https://github.com/breannansmith/scisim/blob/b1397b5504e89e9fefde55e853f344fd79fb1824/scisim/HDF5File.h#L144) at some source code places.
I suggest to reuse a more specialised data type like [the exception "std::runtime_error"](http://en.cppreference.com/w/cpp/error/runtime_error) there. How do you think about to [extend the exception class hierarchy](http://www.cplusplus.com/doc/tutorial/exceptions/#standard_exceptions) eventually?

**Comment by breannansmith** [comment]
I agree with your suggestion, the treatment of errors is currently a bit ad-hoc. I plan to fix this issue.
